GenerateLCParams/handler.ts

```diff
@@ -0,0 +1,164 @@
+
+// TODO Refactor after other PR merge
+const ValidRetrievedLolliPopPubKeys = t.intersection([
+  RetrievedLolliPopPubKeys,
+  t.type({
+    status: t.literal(PubKeyStatusEnum.VALID)
+  })
+]);
+
+type ValidRetrievedLolliPopPubKeys = t.TypeOf<
+  typeof ValidRetrievedLolliPopPubKeys
+>;
+
+/**
+ * Type of a GenerateLCParams handler
+ */
+
+type IGenerateLCParamsHandler = (
+  context: Context,
+  assertionRef: AssertionRef,
+  payload: GenerateLcParamsPayload
+) => Promise<
+  | IResponseSuccessJson<LcParams>
+  | IResponseErrorValidation
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+>;
+/**
+ * Handles requests for generating Lollipop Consumer required params.
+ */
+export const GenerateLCParamsHandler = (
+  lollipopKeysModel: LolliPOPKeysModel,
+  expireGracePeriodInDays: NonNegativeInteger,
+  authJwtGenerator: ReturnType<typeof getGenerateAuthJWT>
+): IGenerateLCParamsHandler => async (
+  _,
+  assertionRef,
+  payload
+): ReturnType<IGenerateLCParamsHandler> =>
+  pipe(
+    lollipopKeysModel.findLastVersionByModelId([assertionRef]),
+    TE.mapLeft(e =>
+      ResponseErrorInternal(
+        `Cannot query for assertionRef on CosmosDB|ERROR=${JSON.stringify(e)}`
+      )
+    ),
+    TE.chainW(
+      TE.fromOption(() =>
+        ResponseErrorNotFound("AssertionRef not found", "Not Found")
+      )
+    ),
+    TE.chainW(
+      flow(
+        ValidRetrievedLolliPopPubKeys.decode,
+        E.mapLeft(() => ResponseErrorForbiddenNotAuthorized),
+        E.chain(
+          E.fromPredicate(
+            usedPubKeyDocument =>
+              usedPubKeyDocument.expiredAt.getTime() >
+              dateUtils.addDays(new Date(), -expireGracePeriodInDays).getTime(),
+            () => ResponseErrorForbiddenNotAuthorized
+          )
+        ),
+        TE.fromEither,
+        TE.bindTo("activePubKey"),
+        TE.bindW("lcAuthJwt", () =>
+          pipe(
+            authJwtGenerator({
+              assertionRef,
+              operationId: payload.operation_id
+            }),
+            TE.mapLeft(e =>
+              ResponseErrorInternal(
+                `Cannot generate LC Auth JWT|ERROR=${e.message}`
+              )
+            )
+          )
+        ),
+        TE.map(({ activePubKey, lcAuthJwt }) =>
+          ResponseSuccessJson({
+            assertion_file_name: activePubKey.assertionFileName,
+            assertion_ref: activePubKey.assertionRef,
+            assertion_type: activePubKey.assertionType,
+            expired_at: activePubKey.expiredAt,
+            fiscal_code: activePubKey.fiscalCode,
+            lc_authentication_bearer: lcAuthJwt,
+            pub_key: activePubKey.pubKey,
+            status: activePubKey.status,
+            ttl: pipe(
+              activePubKey.ttl,
+              O.fromNullable,
+              O.chainEitherK(NonNegativeInteger.decode),
+              O.getOrElse(() => 0 as NonNegativeInteger)
+            ),
```

Comment:
This will be resolved on @arcogabbo's PR
https://github.com/pagopa/io-functions-lollipop/blob/8a0f0a4bedbce3268c025dd5223ebf1fef847a3b/model/lollipop_keys.ts#L100